__tests__/mark_utxos_selected_as_input.test.js

```diff
@@ -0,0 +1,179 @@
+      .send({ txHex, mark_as_used: 123 })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+
+    response = await TestUtils.request
+      .put('/wallet/utxos-selected-as-input')
+      .send({ txHex, mark_as_used: 'abc' })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should fail if ttl is not a number', async () => {
+    const response = await TestUtils.request
+      .put('/wallet/utxos-selected-as-input')
+      .send({ txHex, ttl: '123a' })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should fail if txHex is an invalid transaction', async () => {
+    const response = await TestUtils.request
+      .put('/wallet/utxos-selected-as-input')
+      .send({ txHex: '0123456789abcdef' })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(false);
+    expect(selectSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should mark the inputs as selected on the storage', async () => {
+    const response = await TestUtils.request
+      .put('/wallet/utxos-selected-as-input')
+      .send({ txHex })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+
+    expect(selectSpy).toHaveBeenCalledTimes(3);
+    expect(selectSpy).toHaveBeenCalledWith(
+      { index: 0, txId: '5db0a8c77f818c51cb107532fc1a36785adfa700d81d973fd1f23438b2f3dd74' },
+      true,
+      undefined,
+    );
+  });
+
+  it('should mark the inputs as selected on the storage with ttl', async () => {
+    const response = await TestUtils.request
+      .put('/wallet/utxos-selected-as-input')
+      .send({ txHex, ttl: 123 })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+
+    expect(selectSpy).toHaveBeenCalledTimes(3);
+    expect(selectSpy).toHaveBeenCalledWith(
+      { index: 0, txId: '5db0a8c77f818c51cb107532fc1a36785adfa700d81d973fd1f23438b2f3dd74' },
+      true,
+      123,
+    );
+  });
+
+  it('should mark the inputs as selected on the storage with mark false', async () => {
+    const response = await TestUtils.request
+      .put('/wallet/utxos-selected-as-input')
+      .send({ txHex, mark_as_used: false })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+
+    expect(selectSpy).toHaveBeenCalledTimes(3);
+    expect(selectSpy).toHaveBeenCalledWith(
+      { index: 0, txId: '5db0a8c77f818c51cb107532fc1a36785adfa700d81d973fd1f23438b2f3dd74' },
+      false,
+      undefined,
+    );
+  });
+
+  it('should mark the inputs as selected on the storage with mark true', async () => {
```

Comment:
[question] This is the default behavior, right?

Reply:
Yes, but this test is to check that the behavior does not change when we send `mark_as_used` as `true`